File: s3JavaScriptSDK/sample/WebsiteSerialTesting.js
function websiteSerialTesting (){


	return Q.fcall(function (){
		var deferred = Q.defer();
		console.log('-----------------------------------------');
		console.log('websiteSerialTesting begin!');
		deferred.resolve();
		return deferred.promise;
	})
	.then(function(){
		var params = {
				  Bucket: testBucketName, /* required */
				  WebsiteConfiguration: { /* required */
				    ErrorDocument: {
				      Key: 'error.html' /* required */
				    },
				    IndexDocument: {
				      Suffix: 'index.html' /* required */
				    },
				  },
				};
		return putBucketWebsite(params, s3);
	})
	.then(function(){
		var params = {
				  Bucket: testBucketName /* required */
		};
		return getBucketWebsite(params, s3);
	})
	.then(function(){
		var params = {
				  Bucket: testBucketName /* required */
		};
		return deleteBucketWebsite (params, s3)
	});

		
	// var params = {
	//   Bucket: testBucketName, /* required */
	//   WebsiteConfiguration: { /* required */
	//     ErrorDocument: {
	//       Key: 'error.html' /* required */
	//     },
	//     IndexDocument: {
	//       Suffix: 'index.html' /* required */
	//     },
	    
	//   },
	//   // ContentMD5: 'STRING_VALUE'
	// };
	// s3.putBucketWebsite(params, function(err, data) {
	//   if (err) console.log(err, err.stack); // an error occurred
	//   else     {
	//   	console.log(data);
	//   	getAndDeleteBucketWebsite(testBucketName);
	//   }           // successful response
	// });
}




function putBucketWebsite(params, s3)
{
	var deferred = Q.defer();
	s3.putBucketWebsite(params, function(err, data) {
		  if (err) console.log(err, err.stack); // an error occurred
		  else     {
		  	console.log(data);
			deferred.resolve();
		  }           // successful response
	});
	return deferred.promise;
}

function getBucketWebsite(params, s3)
{
	var deferred = Q.defer();

	s3.getBucketWebsite(params, function(err, data) {
		  if (err) console.log(err, err.stack); // an error occurred
		  else     {
		  	console.log(data);
			deferred.resolve();
		  }           // successful response
	});
	return deferred.promise;
}


function deleteBucketWebsite (params, s3){
	var deferred = Q.defer();

	s3.deleteBucketWebsite(params, function(err, data) {
	  if (err) console.log(err, err.stack); // an error occurred
	  else     {
		  	console.log(data);
			deferred.resolve();
	  }
	});
	return deferred.promise;
}


// function getAndDeleteBucketWebsite (bucketName){
// 	/* body... */
// 	var params = {
// 	  Bucket: bucketName /* required */
// 	};
// 	s3.getBucketWebsite(params, function(err, data) {
// 	  if (err) console.log(err, err.stack); // an error occurred
// 	  else     {
// 	  	console.log(data);
// 	  	deleteBucketWebsite (bucketName);
// 	  }           // successful response
// 	});
// }




// function deleteBucketWebsite (bucketName){
// 	var params = {
// 	  Bucket: bucketName /* required */
// 	};
// 	s3.deleteBucketWebsite(params, function(err, data) {
// 	  if (err) console.log(err, err.stack); // an error occurred
// 	  else     console.log(data);           // successful response
// 	});
// }

File: s3RubySDK/sample/WebsiteSerialTesting.rb
def websiteserialtesting(s3, buckets, creds)
  cleanup
  
  
  resp = s3.create_bucket({
    acl: "public-read", # accepts private, public-read, public-read-write, authenticated-read
    bucket: buckets[0], # required
    create_bucket_configuration: {
      location_constraint: "ap-northeast-1", # accepts EU, eu-west-1, us-west-1, us-west-2, ap-southeast-1, ap-southeast-2, ap-northeast-1, sa-east-1, cn-north-1, eu-central-1
    }
    })
  
  
  
  resp = s3.put_bucket_website({
    bucket: buckets[0], # required
#    content_md5: "ContentMD5",
    website_configuration: { # required
      error_document: {
        key: "error.html", # required
      },
      index_document: {
        suffix: "index.html", # required
      },
#      redirect_all_requests_to: {
#        host_name: "example.com", # required
#        protocol: "https", # accepts http, https
#      },
#      routing_rules: [
#        {
#          condition: {
#            http_error_code_returned_equals: "HttpErrorCodeReturnedEquals",
#            key_prefix_equals: "KeyPrefixEquals",
#          },
#          redirect: { # required
#            host_name: "example.com",
#            http_redirect_code: "HttpRedirectCode",
#            protocol: "http", # accepts http, https
#            replace_key_prefix_with: "ReplaceKeyPrefixWith",
#            replace_key_with: "ReplaceKeyWith",
#          },
#        },
#      ],
    },
  })
  
  resp = s3.get_bucket_website({
    bucket: buckets[0], # required
  })
      
#  puts resp.redirect_all_requests_to.host_name #=> String
#  puts resp.redirect_all_requests_to.protocol #=> String, one of "http", "https"
  puts resp.index_document.suffix #=> String
  puts resp.error_document.key #=> String
#  puts resp.routing_rules[0].condition.http_error_code_returned_equals #=> String
#  puts resp.routing_rules[0].condition.key_prefix_equals #=> String
#  puts resp.routing_rules[0].redirect.host_name #=> String
#  puts resp.routing_rules[0].redirect.http_redirect_code #=> String
#  puts resp.routing_rules[0].redirect.protocol #=> String, one of "http", "https"
#  puts resp.routing_rules[0].redirect.replace_key_prefix_with #=> String
#  puts resp.routing_rules[0].redirect.replace_key_with #=> String

  resp = s3.delete_bucket_website({
    bucket: buckets[0], # required
  })
  
  # clean all buckets
  cleanup
end